ir/instr.h

```diff
@@ -569,19 +682,26 @@ class Memset final : public Instr {
 };
 
 
-class Memcpy final : public Instr {
+class Memcpy final : public MemInstr {
   Value *dst, *src, *bytes;
   unsigned align_dst, align_src;
   bool move;
 public:
   Memcpy(Value &dst, Value &src, Value &bytes,
          unsigned align_dst, unsigned align_src, bool move)
-    : Instr(Type::voidTy, "memcpy"), dst(&dst), src(&src), bytes(&bytes),
+    : MemInstr(Type::voidTy, "memcpy"), dst(&dst), src(&src), bytes(&bytes),
             align_dst(align_dst), align_src(align_src), move(move) {}
 
   Value& getBytes() const { return *bytes; }
   unsigned getSrcAlign() const { return align_src; }
   unsigned getDstAlign() const { return align_dst; }
+
+  uint64_t getMaxAllocSize() const override { return 0; }
```

Comment:
please don't define virtual methods in the headers. this increases compile time a lot.